src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -0,0 +1,1753 @@
+      }
+
+      // 8 bits int
+      case Types.TINYINT: {
+        if (defn.isSignedNumber()) {
+          return new ColumnConverter() {
+            @Override
+            public Object convert(ResultSet rs) throws SQLException {
+              return rs.getByte(col);
+            }
+          };
+        } else {
+          return new ColumnConverter() {
+            @Override
+            public Object convert(ResultSet rs) throws SQLException {
+              return rs.getShort(col);
+            }
+          };
+        }
+      }
+
+      // 16 bits int
+      case Types.SMALLINT: {
+        if (defn.isSignedNumber()) {
+          return new ColumnConverter() {
+            @Override
+            public Object convert(ResultSet rs) throws SQLException {
+              return rs.getShort(col);
+            }
+          };
+        } else {
+          return new ColumnConverter() {
+            @Override
+            public Object convert(ResultSet rs) throws SQLException {
+              return rs.getInt(col);
+            }
+          };
+        }
+      }
+
+      // 32 bits int
+      case Types.INTEGER: {
+        if (defn.isSignedNumber()) {
+          return new ColumnConverter() {
+            @Override
+            public Object convert(ResultSet rs) throws SQLException {
+              return rs.getInt(col);
+            }
+          };
+        } else {
+          return new ColumnConverter() {
+            @Override
+            public Object convert(ResultSet rs) throws SQLException {
+              return rs.getLong(col);
+            }
+          };
+        }
+      }
+
+      // 64 bits int
+      case Types.BIGINT: {
+        return new ColumnConverter() {
+          @Override
+          public Object convert(ResultSet rs) throws SQLException {
+            return rs.getLong(col);
+          }
+        };
+      }
+
+      // REAL is a single precision floating point value, i.e. a Java float
+      case Types.REAL: {
+        return new ColumnConverter() {
+          @Override
+          public Object convert(ResultSet rs) throws SQLException {
+            return rs.getFloat(col);
+          }
+        };
+      }
+
+      // FLOAT is, confusingly, double precision and effectively the same as DOUBLE. See REAL
+      // for single precision
+      case Types.FLOAT:
+      case Types.DOUBLE: {
+        return new ColumnConverter() {
+          @Override
+          public Object convert(ResultSet rs) throws SQLException {
+            return rs.getDouble(col);
+          }
+        };
+      }
+
+      case Types.NUMERIC:
+        if (mapNumerics) {
+          int precision = defn.precision();
+          if (precision < 19 && defn.scale() == 0) { // integer
+            if (precision > 9) {
+              return new ColumnConverter() {
+                @Override
+                public Object convert(ResultSet rs) throws SQLException {
+                  return rs.getLong(col);
+                }
+              };
```

Comment:
While most of these cases don't have any custom logic, here's an example of where the overhead of looking at the precision and scale for a column becomes too expensive to do for every field in every row, and where returning a function is far superior.